### DISCUSSION TITLE
`autoincrement()` with composite key

### DISCUSSION THREAD

**Discussion by yorkeJohn** [prisma/prisma#25733] [Q&A]
### Question

How can I have multiple auto-incremented sequences within a composite key?

e.g.
```prisma
model WorkflowRun {
  id         Int      @default(autoincrement())
  workflow   Workflow @relation(fields: [workflowId], references: [id])
  workflowId String
  
  @@id([workflowId, id])
}
```
Insert some data
```ts
  await prisma.workflowRun.create({data: {workflowId: 1}})
  await prisma.workflowRun.create({data: {workflowId: 1}})
  await prisma.workflowRun.create({data: {workflowId: 2}})
  await prisma.workflowRun.create({data: {workflowId: 2}})

  {id: 1, workflowId: 1}
  {id: 2, workflowId: 1}
  {id: 3, workflowId: 2} // should have id 1
  {id: 4, workflowId: 2} // should have id 2
```

**Comment by ludralph** [comment]
Hi @yorkeJohn 

Thank you for raising this question. 

Prisma does not currently support having multiple auto-incremented sequences within a composite key as you described. The autoincrement() feature in Prisma is designed to work with a single sequence for the entire table, not separate sequences per group. You can open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) and document your usecase so that our engineers can take a look.

**Reply by ineshbose** [reply]
This is the feature request I found: #22440
